### ISSUE TITLE
Make the declaration of own work digitally signable

### ISSUE THREAD

**Issue by hochleitner** [Digital-Media/HagenbergThesis#101]
It might be a good idea, to make the declaration of own work ("eidesstattliche Erklärung") digitally signable so digital copies have some sort of dependability.

The packages `digisig` or `eforms` might work, as suggested in https://tex.stackexchange.com/questions/51090/how-do-i-create-a-pdf-file-that-can-be-digitally-signed.

**Comment by imagingbook** [comment]
We need to consider though that, once digitally signed, a document cannot be edited (or commented)  any more (at least not without inappropriate tricks). This would make it useless for giving feedback or adding notes.

**Comment by hochleitner** [comment]
So let's assume that students do not sign it until they actually hand it in.

Otherwise, we could enable the creation of a digitally signable field with an additional parameter for the class. I think this could be a viable solution that is also backward compatible.

**Comment by imagingbook** [comment]
Few more changes (in ``main.tex``, ``hgbthesis.cls``): 
* the user command changed to ``\AddSignatureField``;
* the internal boolean control flag is ``hgb@addsignatureflag``;
* the internal macro which inserts the signature field is ``\hgb@signaturefield``.

**Comment by imagingbook** [comment]
@hochleitner : Are we ready to merge this?

**Comment by hochleitner** [comment]
Sorry, totally forgot about this. Will test this weekend and report back.

**Comment by hochleitner** [comment]
Okay, I checked everything out and it works. There's a few things that came up during testing for me.

1. Did you manage to get an official ID provider to work with Acrobat? Because it only allows either signature devices or files and the Austrian "Handy Signatur" is neither. So I could only create a self-signed certificate, which is okay I guess? Question is, do we need proper verification?
2. I also tried to sign the document with my "Handy Signatur". I tried two ways. First, the local PDF-Over application from A-Trust (https://www.buergerkarte.at/software-pdf-signatur-handy.html) and then the Online-Adobe-signature from A-Trust (also on https://www.buergerkarte.at/software-pdf-signatur-handy.html, bit further down). Those work well and generate an official signature stamp that can also be verified. The issue with those is just that they do not respect the signature field. This "stamp" can be put anywhere, so if you use one of these, we wouldn't really need all of this.

So the question remains - what do we want (or need?) on this page? Just the student's signature or a proper digital ID that can also be verified?
